addons/docs/src/mdx/mdx-compiler-plugin.js

```diff
@@ -52,37 +52,44 @@ function genStoryExport(ast, context) {
   const statements = [];
   const storyKey = getStoryKey(storyName, context.counter);
 
-  let body = ast.children.find(n => n.type !== 'JSXText');
+  const bodyNodes = ast.children.filter(n => n.type !== 'JSXText');
   let storyCode = null;
-
-  if (!body) {
+  let storyVal = null;
+  if (!bodyNodes.length) {
     // plain text node
     const { code } = generate(ast.children[0], {});
     storyCode = `'${code}'`;
+    storyVal = `() => (
+      ${storyCode}
+    )`;
   } else {
-    if (body.type === 'JSXExpressionContainer') {
-      // FIXME: handle fragments
-      body = body.expression;
+    const bodyParts = bodyNodes.map(bodyNode => {
```

Comment:
Would be simpler, and even more correct, if we just wrapped the child nodes in a fragment?

Reply:
Can you please explain, not sure i follow where you have it in mind?

Reply:
Pseudocode since I'm on my phone:

```
if(len(Non-text children) > 1):
 story.children = wrapInFragment(story.children)
```

Reply:
Hmm I need to think about the non-react case too 🤔